model/private_subnet.rb

```diff
@@ -30,6 +30,16 @@ def hyper_tag_name(project)
 
   include Authorization::TaggableMethods
 
+  def connected_subnets
+    PrivateSubnet.where(
+      id: DB[:connected_subnet].where(subnet_id_1: id).or(subnet_id_2: id).select(Sequel.case({{subnet_id_1: id} => :subnet_id_2}, :subnet_id_1)).all.map(&:values).flatten
+    ).all
+  end
+
+  def all_nics
+    nics + connected_subnets.flat_map(&:nics)
+  end
```

Comment:
feels like this could be a join rather than detoasting a bunch of stuff. also, `flat_map` rather than map + flatten.